src/engine/shared/network_token.cpp

```diff
@@ -1,23 +1,17 @@
 /* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 
+#include <base/hash_ctxt.h>
 #include <base/math.h>
 #include <base/system.h>
 
-#include <engine/external/md5/md5.h>
-
 #include "network.h"
 
 static unsigned int Hash(char *pData, int Size)
 {
-	md5_state_t State;
-	unsigned int aDigest[4]; // make sure this is 16 byte
-
-	md5_init(&State);
-	md5_append(&State, (const md5_byte_t *)pData, Size);
-	md5_finish(&State, (md5_byte_t *)aDigest);
+	MD5_DIGEST Digest = md5(pData, Size);
 
-	return (aDigest[0] ^ aDigest[1] ^ aDigest[2] ^ aDigest[3]);
+	return (Digest.data[0] ^ Digest.data[1] ^ Digest.data[2] ^ Digest.data[3]);
```

Comment:
Should use uint* here, like before. `Digest.data` is uchar*.

Reply:
Oh, missed that you put the digest into an integer array.
I don't see a good way of doing that barring a `reinterpret_cast`. If you are okay with that I can commit it.
Or I can do something like `data[0] + data[1]<<0x8 + data[2] << 0x10 + data[3] << 0x18`

Reply:
For best readability it might be best to cast `Digest.data` to a uint* and then do the ^-operation with it.